db/stats_history.h

```diff
@@ -0,0 +1,96 @@
+// Copyright (c) 2011-present, Facebook, Inc.  All rights reserved.
+// //  This source code is licensed under both the GPLv2 (found in the
+// //  COPYING file in the root directory) and Apache 2.0 License
+// //  (found in the LICENSE.Apache file in the root directory).
+// // Copyright (c) 2011 The LevelDB Authors. All rights reserved.
+// // Use of this source code is governed by a BSD-style license that can be
+// // found in the LICENSE file. See the AUTHORS file for names of contributors.
+
+#pragma once
+
+#include <map>
+#include <string>
+
+// #include "db/db_impl.h"
+#include "rocksdb/statistics.h"
+#include "rocksdb/status.h"
+
+namespace rocksdb {
+
+class DBImpl;
+
+enum AggregationOps : uint32_t {
+  AVG = 0,
+  SUM,
+  MIN,
+  MAX,
+  COUNT,
+  COUNT_DISTINCT,
+  P25,
+  P50,
+  P75,
+  P90,
+  P99,
+
+  AGGREGATION_ENUM_MAX,
+};
+
+// Options that controls GetStatsHistory calls
+struct GetStatsOptions {
+  uint64_t start_time;
+  uint64_t end_time;
+  // list of counters to query, call GetSupportedStatsCounters to get full list
+  std::vector<Tickers> tickers;
+  // aggregation operation like avg, sum, min, max, count, percentiles, etc
+  AggregationOps aggr_ops;
```

Comment:
What is it for?

Reply:
this is for user to specify what kind of aggregation operation to apply on the data, for example, average, sum, min/max, p90, etc. it's not implemented or used anywhere in the initial version. 
One thing that's not clear is whether to have two separate accessor for raw StatsRecord and aggregated stats. For example, `GetStatsHistory` is called for 1 hour interval, certain counters, aggregation operation of average and p90. What is the best way to return the data? I think we can have a StatsRecord() to return the raw data together with timestamp and a AggregatedStats() to return a single aggregated result? Do we need to support aggregation operations that returns more than a single floating point value? How about aggregation operations that takes multiple counters?

Reply:
If it is not implemented or used. I suggest we remove it for now. We can always add it back later.

Reply:
@siying got a question for tickers: what form of ticker should users provide? 
Currently each ticker have two forms, one is of type enum and another is std::string. For example, `Tickers::BLOCK_CACHE_MISS` or `"rocksdb.block.cache.miss"`, and the mapping is maintained in `TickersNameMap`. Is it more user friendly to ask users to provide the latter and add lookup functions to return the enum type to do the filtering before returning the results?